dll/win32/shell32/CDefView.cpp

```diff
@@ -2582,6 +2617,14 @@ HRESULT WINAPI CDefView::DestroyViewWindow()
 {
     TRACE("(%p)\n", this);
 
+    if (m_hUpdateStatusbarThread)
+    {
+        HANDLE hOldThread = m_hUpdateStatusbarThread;
+        m_hUpdateStatusbarThread = NULL;
+        ::WaitForSingleObject(hOldThread, INFINITE);
```

Comment:
Will this block browsing until the thread is killed?

Reply:
Assigning `NULL` to `m_hUpdateStatusbarThread` will terminate the thread immediately.

Reply:
Not immediately, because then there would be no need for a thread (if all operations in the thread return immediately).